### PR TITLE
More tests, use fns from utils and rework core

### PR DESCRIPTION
**Pull request by mitchellhamilton**
**What**:

**Why**:

**How**:

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Code complete

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -236,21 +221,22 @@ function selector (id: string, path: string = '') {
...
-    .split(',')
-    .map(
-      x =>
-        x.indexOf('&') >= 0 ? x.replace(/&/gm, `.css-${id}`) : `.css-${id}${x}`
-    )
-    .join(',')
+  let x = map(
+    path.split(','),
+    x =>
+      x.indexOf('&') >= 0 ? x.replace(/&/gm, `.css-${id}`) : `.css-${id}${x}`
```
Comment (tkh44) [inline src/index.js]: We should probably hoist this callback and the regex inside.

File: src/index.js [hunk 2]
```diff
@@ -342,9 +328,11 @@ function joinSelectors (a, b) {
   let as = map(a.split(','), a => (!(a.indexOf('&') >= 0) ? '&' + a : a))
   let bs = map(b.split(','), b => (!(b.indexOf('&') >= 0) ? '&' + b : b))
 
-  return bs
-    .reduce((arr, b) => arr.concat(as.map(a => b.replace(/&/g, a))), [])
-    .join(',')
+  return reduce(
+    bs,
+    (arr, b) => arr.concat(map(as, a => b.replace(/&/g, a))),
```
Comment (tkh44) [inline src/index.js]: We could hoist this up as well.

File: src/index.js (hunk 1 above)
Comment (mitchellhamilton) [inline src/index.js]: We can't because it uses `id`

File: src/index.js (hunk 2 above)
Comment (mitchellhamilton) [inline src/index.js]: We can't because it uses `a`

Comment (mitchellhamilton) [inline src/index.js]: wait, i might be wrong

Comment (mitchellhamilton) [inline src/index.js]: We can't hoist it because it uses `as`

**Pull request merged**